### DISCUSSION TITLE
Supabase authentication or connection

### DISCUSSION THREAD

**Discussion by LautaroZanzot** [micropython/micropython#15266] [ESP8266]
I am doing a project with the esp8266 module, and the data has to be sent to a db in this case it is supabase, but when I make the post with urequest it gives the error 401
`def send_data(tempC, defrost, contrl, bateria):

    json_data = ujson.dumps({
        "temp": str(tempC),
        "defrost": str(defrost),
        "arranque_comp": str(contrl),
        "bateria": str(bateria),
        "idvigia": str(IdVigia)
    })
    headers = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + API_KEY
    }
    response = request.post(API_URL + "/" + TableName, headers=headers, data=json_data)
    if response.status_code == 401:
        print("Error 401: Unauthorized")
        print(API_URL + "/" + TableName, headers, json_data)
        print("Reason:", response.reason)
        print("Text:", response.text)
    else:
        print("HTTP response code:", response.status_code)
        print(json_data)
        print(API_URL + "/" + TableName, headers, json_data)
`

all the prints are
`Error 401: Unauthorized
https://...supabase.com/signals {'Authorization': 'Bearer eyJh...3gc', 'Content-Type': 'application/json'} {"temp": "[]", "defrost": "0", "bateria": "0", "idvigia": "1", "arranque_comp": "0"}
Reason: b'Unauthorized'
Text: {"message":"No API key found in request","hint":"No `apikey` request header or url param was found."}`

**Comment by glenn20** [comment]
Have you tested accessing your db from regular cpython? I suspect this is not a micropython problem. The returned (very helpful) message indicates you need to set the "apikey" header:

```py
  headers = {
      "Content-Type": "application/json",
      "apikey": API_KEY,
      "Authorization": "Bearer " + API_KEY
}
```

I dont know how your supabase auth works, it may be that you have to use the apikey to access the instance, but you may then have to authenticate a user session and access_token before inserting data.
